### PR TITLE
Deploy only specific component

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
The functionality seems ok, but there are no unit tests for the deploy step to verify that when a list of components is specified, they are actually preserved from current active deployment.

File: pipeline-runner/steps/deploy.go [hunk 1]
```diff
@@ -121,6 +134,60 @@ func (cli *DeployStepImplementation) deployToEnv(appName, env string, pipelineIn
 	return nil
 }
 
+func (cli *DeployStepImplementation) validate(pipelineInfo *model.PipelineInfo, env string) error {
```
[the remark this answers is not in the document]
Comment (satr) [reply pipeline-runner/steps/deploy.go]: Moved

File: pipeline-runner/steps/deploy.go [hunk 2]
```diff
@@ -91,8 +91,13 @@ func (cli *DeployStepImplementation) deployToEnv(appName, env string, pipelineIn
 		return err
 	}
 
-	radixDeployment, err := deployment.ConstructForTargetEnvironment(
+	activeRadixDeployment, err := cli.GetKubeutil().GetActiveDeployment(utils.GetEnvironmentNamespace(appName, env))
```
Comment (nilsgstrabo) [inline pipeline-runner/steps/deploy.go:94]: Use existing function getCurrentRadixDeployment. This function ignores Forbidden errors in cases where a new environment is added to radixconfig, or radixconfig is applied for the first time, and the RA controller has not yet finished reconciling RBAC

Comment (satr) [reply pipeline-runner/steps/deploy.go]: Fixed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**